src/modules/select/wallets/ledger.ts

```diff
@@ -185,27 +200,54 @@ async function ledgerProvider(options: {
 
   async function createTransport() {
     try {
-      transport = LedgerTransport
-        ? await LedgerTransport.create()
-        : await TransportU2F.create()
-
-      eth = new Eth(transport)
-
       const observer = {
         next: (event: any) => {
           if (event.type === 'remove') {
             disconnect()
           }
         },
-        error: () => {},
+        error: (error: any) => {
+          throw new Error(error)
+        },
         complete: () => {}
       }
 
-      LedgerTransport
-        ? LedgerTransport.listen(observer)
-        : TransportU2F.listen(observer)
+      if (CustomLedgerTransport) {
+        transport = await CustomLedgerTransport.create()
+        transportSubscription = CustomLedgerTransport.listen(observer)
+      } else {
+        if (
+          os.name === 'Windows' &&
+          parseInt(os.versionName) >= 11 &&
+          (browser.name === 'Internet Explorer' || browser.name === 'Firefox')
+        ) {
+          throw new Error(
+            `OS: ${os.name} ${os.versionName} and Browser: ${browser.name} are not compatible with Ledger wallets, please switch to Chrome browser to continue.`
+          )
+        } else if (
+          (os.name === 'macOS' || os.name === 'Linux') &&
+          (browser.name === 'Firefox' || browser.name === 'Safari')
+        ) {
+          const { default: TransportU2F } = await import(
+            '@ledgerhq/hw-transport-u2f'
+          )
+
+          transport = await TransportU2F.create()
+        } else {
+          const { default: TransportWebUsb } = await import(
+            '@ledgerhq/hw-transport-webusb'
+          )
```

Comment:
is this some cool voodoo to reduce initial page load time?

Reply:
Haha yep, dynamic imports will only load the module when and if it is needed rather than all at once on initial load, reducing data usage and load time - https://medium.com/rollup/rollup-now-has-code-splitting-and-we-need-your-help-46defd901c82